Core/Versioning/KspVersionCriteria.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace CKAN.Versioning
+{
+    public class KspVersionCriteria
+    {
+        private List<KspVersion> versions = new List<KspVersion> ();
+
+        public KspVersionCriteria (KspVersion v)
+        {
+            this.versions.Add (v);
+        }
+
+        public KspVersionCriteria(KspVersion v, List<KspVersion> compatibleVersions)
```

Comment:
Rather than having two constructors how about just a single one with a variadic parameter:
```C#
public KspVersionSet(params KspVersion[] versions) { }
```